Lib/pdb.py

```diff
@@ -142,8 +142,10 @@ def check(self):
             print('Error:', self.orig, 'is a directory')
             sys.exit(1)
 
-        # Replace pdb's dir with script's dir in front of module search path.
-        sys.path[0] = os.path.dirname(self)
+        # Replace pdb's dir with script's dir in front of module search path
+        # if safe_path is not set, otherwise sys.path[0] is not pdb's dir
+        if not getattr(sys.flags, 'safe_path', None):
```

Comment:
is `safe_path` not guaranteed to exist? the test does `sys.flags.safe_path`.

Reply:
I derived this from the original author, but from the source code it seems like `sys.flags.safe_path` should always exist after 3.11. I'll change it here, maybe together with the comment after we decide what to do.

Reply:
I don’t remember exactly, but I probably added the check because I was not sure whether it’s okay to depend on ≥ 3.11. If you know that it is, go ahead and remove it.